test/unit/CodeGenerator/NameGeneratorTest.php

```diff
@@ -0,0 +1,791 @@
+                                                    ],
+
+                                                    'responseMarkersInterfaceNames' => [
+                                                        'extends' => null,
+                                                        'fileName' => 'SomeCustomOperationIdResponse.php',
+                                                        'filePath' => str_replace(['/', '\\'], DIRECTORY_SEPARATOR, '/Some/Custom/Path/Apis/SomeCustomNamespace/SomeCustomOperationId/Dto/Response'),
+                                                        'className' => 'SomeCustomOperationIdResponse',
+                                                        'namespace' => 'Some\Custom\Namespace\Apis\SomeCustomNamespace\SomeCustomOperationId\Dto\Response',
+                                                    ],
+                                                ],
+                                            ],
+                                            'expected' => [
+                                                'requestHandlerInterface' => [
+                                                    'methodName' => 'someCustomOperationId',
+                                                    'methodDescription' => 'SomeCustomSummary',
+                                                    'fileName' => 'SomeCustomOperationId.php',
+                                                    'filePath' => str_replace(['/', '\\'], DIRECTORY_SEPARATOR, '/Some/Custom/Path/Apis/SomeCustomNamespace/SomeCustomOperationId'),
+                                                    'namespace' => 'Some\Custom\Namespace\Apis\SomeCustomNamespace\SomeCustomOperationId',
+                                                    'className' => 'SomeCustomOperationId',
+                                                ],
+                                            ],
+                                            'hasMakersInterface' => true,
+                                        ],
+                                    ],
+                                ],
+                            ],
+                        ],
+                    ]
+                ),
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed[] $payload
+     *
+     * @dataProvider setAllNamesAndPathsProvider
+     */
+    public function testSetAllNamesAndPaths(array $payload): void
+    {
+        $namingStrategy = new DefaultNamingStrategy(
+            new ReservedWords(),
+            $payload['namingStrategy']['rootNamespace'],
+            $payload['namingStrategy']['languageLevel']
+        );
+
+        $httpStatus = new Httpstatus($payload['httpStatus']['statusArray']);
+
+        /** @var SpecificationDefinition[]|array $specifications */
+        $specifications = array_map(static function (array $payload): SpecificationDefinition {
```

Comment:
This whole block can be moved to a private method of the test, to make the test method itself easier to read.